src/main/java/uk/gov/register/presentation/view/EntryListView.java

```diff
@@ -5,34 +5,43 @@
 import uk.gov.register.presentation.config.PublicBody;
 import uk.gov.register.presentation.dao.Entry;
 import uk.gov.register.presentation.representations.CsvRepresentation;
+import uk.gov.register.presentation.resource.NewPagination;
 import uk.gov.register.presentation.resource.Pagination;
 import uk.gov.register.presentation.resource.RequestContext;
 
+import java.util.Collection;
 import java.util.List;
 import java.util.Optional;
 
 public class EntryListView extends CsvRepresentationView {
+    private NewPagination newPagination;
     private Pagination pagination;
-    private List<Entry> entries;
+    private Collection<Entry> entries;
 
     public EntryListView(RequestContext requestContext, Pagination pagination, PublicBody custodian, Optional<GovukOrganisation.Details> custodianBranding, List<Entry> entries) {
         super(requestContext, custodian, custodianBranding, "entries.html");
         this.pagination = pagination;
         this.entries = entries;
     }
 
+    public EntryListView(RequestContext requestContext, NewPagination newPagination, PublicBody custodian, Optional<GovukOrganisation.Details> custodianBranding, Collection<Entry> entries, String template) {
+        super(requestContext, custodian, custodianBranding, template);
+        this.newPagination = newPagination;
+        this.entries = entries;
+    }
+
     @JsonValue
-    public List<Entry> getEntries() {
+    public Collection<Entry> getEntries() {
         return entries;
     }
 
     @SuppressWarnings("unused, used from templates")
-    public Pagination getPagination() {
-        return pagination;
+    public NewPagination getPagination() {
+        return newPagination;
     }
 
```

Comment:
Yes, I would like to remove the Old pagination object so didn't do anymore work. Have you had any discussion about the questions above?